test_sdk_installation: Reduce code duplication

gunnarx opened the pull request:
I suppose this is the solution to issue #6 

I think the duplication is very minor since there are only two tests, but in any case:

Counting non-comment lines:
7+7 lines per test was reduced to 3+3 for each, but then 6 of those lines of course reappear in the new common function.  So in total a reduction of 2 lines ;-)

Of course it's still clearer now, and as additional tests are written the gain can be larger over time.

Commit:
test_sdk_installation: Reduce code duplication

Signed-off-by: Gunnar Andersson <gandersson@genivi.org>

GordanM commented:
@TSAnsari can you please take a look at this PR? :)

DunderRoffe reviewed (commented):
I have a vague memory that @rpannek tried this but it did not work... Although I might be wrong in which case I approve :slightly_smiling_face:

rpannek commented:
Hm I remember it also not being called or something, but perhaps I had it in a different way? I'll need to try that manually because i don't think the Pelux team connected it to a automatic test build yet.

thenor commented:
This PR seems to have been forgotten. Did you try this manually @rpannek?

rpannek commented:
No, sorry I totally forgot about it. Sadly I won't really have time to test it any time soon either, perhaps someone else could try it?

sashko commented:
@gunnarx all the tests fail with this patch. I know it's old now, so you decide if you want to rebase or reject it.

gunnarx commented:
Sorry I was out of date.  Rebasing now and we'll see.

gunnarx commented:
Yeah a simple bug in there.  I fixed it and it's now up to date with master.  One test failed for me on both this PR and master which is the can_find_dlt_viewer.py, but I'm not sure I got the test setup correct yet.

sashko commented:
@gunnarx I'm afraid all the tests fail with this patch.

thenor commented:
I think this is a good improvement to the code, but it seems that it doesn't work for some reason.

gunnarx commented:
> @gunnarx I'm afraid all the tests fail with this patch.

Hmm.  Not sure if misunderstanding...  What I tried to say was only that I had an issue with testing preventing me from confirming 100%.   But the "all tests fail" problem was trivial and I believe I fixed it in https://github.com/Pelagicore/pelux-sde/pull/13/commits/38f4afa92c3ad110a0e4f35d698263fd3b6cfa30.  which was pushed yesterday when I wrote the comment.  Just to be clear, did you retest that one or were you commenting based on the original?  

Anyway, here is where I stand now:  Rebased again since I saw updates today.  But I had to go on top of  7bd7a22573dd10e15b5fc4fd5c116576cd92666e (which is just before the Ubuntu 18.04 update). All tests run OK now...

The reason I rebased on 7bd7a22573dd10e15b5fc4fd5c116576cd92666e was that on the latest origin/master from today I have [this issue](https://stackoverflow.com/questions/50573940/issue-with-vagrant-up-with-bento-ubuntu-18-04-on-mac) (but I'm running Linux, not Mac to be clear)

Anyway that's likely some vagrant / guest mismatch issue.  It should be totally independent.  I believe the patch is OK since yesterday.  If you don't mind checking again (or give more info on the error)?

By the way there seemed to be an automated Jenkins job before, but it has not triggered again after I updated?

gunnarx commented:
To explain further, I made this fix yesterday, which was an obvious bug.  It didn't get the right path to tests
```
-        assert os.system("SDK_FILE_NAME=\"{}\" NO_GUI=1 vagrant up".format(stub_script_name)) == 0
+        assert os.system("SDK_FILE_NAME=\"test/stubs/{}\" NO_GUI=1 vagrant up".format(stub_script_name)) == 0
```

thenor commented:
I will pull the code and try it myself again.

gunnarx commented:
Cool.  Hope the automated builds/tests can get set up again too, since I'm starting to get concerned we're all spending too much time on this trivial refactoring.  It's really not terribly important, this ;)  Cheers.

EDIT: I blame @rpannek for creating an issue for refactoring the code in the first place. ;-)  And since I was hacking SDE stuff anyway I jumped in.

sashko commented:
@gunnarx both me and @thenor have tried to run tests with your patch and all the test cases fail. They do run successfully without your patch.

thenor commented:
I think the issue in our CI/CD environment is that py.test-3 was not installed. I will install it now. We have new host machines and unfortunately we have had to go issue by issue.

gunnarx commented:
Well it's a weird feeling when I get a "pass" because it would seem unlikely to be buggy.  Unless of course the script is not used when I test so I am running the tests wrong somehow.

gunnarx commented:
Things you can do is - try to report what is actually failing.  And/or review the patch code and let me know if it does not look correct.

By the way, str.format looks like it was introduced in python3.  It's the only thing that is really unique in the patch... so it must be something there... python versions?

sashko commented:
```
================================================================================================================================== test session starts ===================================================================================================================================
platform linux -- Python 3.6.6, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python3
cachedir: .cache
rootdir: /home/oleksandr/repos/pelux-sde/test, inifile:
collected 11 items                                                                                                                                                                                                                                                                       

test_sdk_installation.py::Test::test_script_inside_vm[can_find_sdk.py] FAILED                                                                                                                                                                                                      [  9%]
test_sdk_installation.py::Test::test_script_inside_vm[can_find_dlt_viewer.py] FAILED                                                                                                                                                                                               [ 18%]
test_sdk_installation.py::Test::test_script_inside_vm[qtcreator_is_configured_to_use_sdk.py] FAILED                                                                                                                                                                                [ 27%]
test_sdk_installation.py::Test::test_script_inside_vm[can_find_template_service_wizard.py] FAILED                                                                                                                                                                                  [ 36%]
test_sdk_installation.py::Test::test_script_inside_vm[check_is_ubuntu.py] FAILED                                                                                                                                                                                                   [ 45%]
test_sdk_installation.py::Test::test_script_inside_vm_qt[qtcreator_is_configured_to_use_sdk_with_qt.py] FAILED                                                                                                                                                                     [ 54%]
test_sdk_installation.py::TestNoQt::test_script_inside_vm[can_find_sdk.py] FAILED                                                                                                                                                                                                  [ 63%]
test_sdk_installation.py::TestNoQt::test_script_inside_vm[can_find_dlt_viewer.py] FAILED                                                                                                                                                                                           [ 72%]
test_sdk_installation.py::TestNoQt::test_script_inside_vm[qtcreator_is_configured_to_use_sdk.py] FAILED                                                                                                                                                                            [ 81%]
test_sdk_installation.py::TestNoQt::test_script_inside_vm[can_find_template_service_wizard.py] FAILED                                                                                                                                                                              [ 90%]
test_sdk_installation.py::TestNoQt::test_script_inside_vm[check_is_ubuntu.py] FAILED                                                                                                                                                                                               [100%]

-------------------------------------------------------------------------------------------------------- generated xml file: /home/oleksandr/repos/pelux-sde/test/test_result.xml --------------------------------------------------------------------------------------------------------
======================================================================================================================================== FAILURES ========================================================================================================================================
______________________________________________________________________________________________________________________ Test.test_script_inside_vm[can_find_sdk.py] _______________________________________________________________________________________________________________________

self = <test_sdk_installation.Test object at 0x7fad6819b898>, script = 'can_find_sdk.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'can_find_sdk.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

___________________________________________________________________________________________________________________ Test.test_script_inside_vm[can_find_dlt_viewer.py] ___________________________________________________________________________________________________________________

self = <test_sdk_installation.Test object at 0x7fad6817cda0>, script = 'can_find_dlt_viewer.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'can_find_dlt_viewer.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

___________________________________________________________________________________________________________ Test.test_script_inside_vm[qtcreator_is_configured_to_use_sdk.py] ____________________________________________________________________________________________________________

self = <test_sdk_installation.Test object at 0x7fad68199be0>, script = 'qtcreator_is_configured_to_use_sdk.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'qtcreator_is_configured_to_use_sdk.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

____________________________________________________________________________________________________________ Test.test_script_inside_vm[can_find_template_service_wizard.py] _____________________________________________________________________________________________________________

self = <test_sdk_installation.Test object at 0x7fad681877f0>, script = 'can_find_template_service_wizard.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'can_find_template_service_wizard.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

_____________________________________________________________________________________________________________________ Test.test_script_inside_vm[check_is_ubuntu.py] _____________________________________________________________________________________________________________________

self = <test_sdk_installation.Test object at 0x7fad68190400>, script = 'check_is_ubuntu.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'check_is_ubuntu.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

______________________________________________________________________________________________________ Test.test_script_inside_vm_qt[qtcreator_is_configured_to_use_sdk_with_qt.py] ______________________________________________________________________________________________________

self = <test_sdk_installation.Test object at 0x7fad681a1e80>, script = 'qtcreator_is_configured_to_use_sdk_with_qt.py'

    @pytest.mark.parametrize("script", scripts_for("qt"))
    def test_script_inside_vm_qt(self, script):
>       run_test_script_inside_vm("qt", script)

test_sdk_installation.py:84: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'qt', script_name = 'qtcreator_is_configured_to_use_sdk_with_qt.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

____________________________________________________________________________________________________________________ TestNoQt.test_script_inside_vm[can_find_sdk.py] _____________________________________________________________________________________________________________________

self = <test_sdk_installation.TestNoQt object at 0x7fad68199a20>, script = 'can_find_sdk.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'can_find_sdk.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

_________________________________________________________________________________________________________________ TestNoQt.test_script_inside_vm[can_find_dlt_viewer.py] _________________________________________________________________________________________________________________

self = <test_sdk_installation.TestNoQt object at 0x7fad68190278>, script = 'can_find_dlt_viewer.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'can_find_dlt_viewer.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

_________________________________________________________________________________________________________ TestNoQt.test_script_inside_vm[qtcreator_is_configured_to_use_sdk.py] __________________________________________________________________________________________________________

self = <test_sdk_installation.TestNoQt object at 0x7fad6817c7f0>, script = 'qtcreator_is_configured_to_use_sdk.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'qtcreator_is_configured_to_use_sdk.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

__________________________________________________________________________________________________________ TestNoQt.test_script_inside_vm[can_find_template_service_wizard.py] ___________________________________________________________________________________________________________

self = <test_sdk_installation.TestNoQt object at 0x7fad68199940>, script = 'can_find_template_service_wizard.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'can_find_template_service_wizard.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

___________________________________________________________________________________________________________________ TestNoQt.test_script_inside_vm[check_is_ubuntu.py] ___________________________________________________________________________________________________________________

self = <test_sdk_installation.TestNoQt object at 0x7fad683ab278>, script = 'check_is_ubuntu.py'

    @pytest.mark.parametrize("script", scripts_for("base"))
    def test_script_inside_vm(self, script):
>       run_test_script_inside_vm("base", script)

test_sdk_installation.py:92: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

script_type = 'base', script_name = 'check_is_ubuntu.py', script_args = ''

    def run_test_script_inside_vm(script_type, script_name, script_args=""):
        command_to_run_in_vm = "python3 {}/{} {}".format(
            scripts_path_on_vm(script_type),
            script_name,
            script_args)
    
        args = ["vagrant", "ssh", "-c", command_to_run_in_vm]
        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        script_exit_code = process.wait()
    
        if script_exit_code != 0:
            print(pipe_to_str(process.stdout))
            print(pipe_to_str(process.stderr), file=sys.stderr)
    
>       assert script_exit_code == 0
E       assert 1 == 0

test_sdk_installation.py:46: AssertionError
---------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------------------------------------------------------------
VM must be created before running this command. Run `vagrant up` first.

=============================================================================================================================== 11 failed in 8.20 seconds ================================================================================================================================
```

gunnarx commented:
Yes I've seen that type of log before it says not much except that the script_exit_code is NOT zero.  Always.

gunnarx commented:
I assume you feel the same as me that this "works on my computer" is getting annoying.  And I wonder if you looked at the code at all or just say "doesn't work..."

Thinking a bit more about this - I think the right thing to do now is to turn over the code responsibility to you.   If you feel reducing this duplication is a useful change then just do it yourself.   (Using python3 or python2 syntax or whatever is the correct way for you). 

If this refactoring is not on your own agenda just move on with what is more important.  It should be easy to see what I **intended** to do at least, and when you feel like it you can repeat it if you want.

I mean is the change is quite trivial, so you can take over to do the change yourself (or not), rather than us speaking more about "my patch".

thenor commented:
@gunnarx I have looked at the code and I totally support the improvement, but at the moment I am stuck at not wanting to integrate since we already have issues in our CI/CD environment and the other option of integrating and fixing what needs to be fixed in this piece of code (if needed) later.

thenor commented:
My conclusion is to do the latter option. We have the issue open still and we will want to solve it, but at the moment we don't have the capacity to look into why this is not working for us. We will keep the code as inspiration. Thank you, @gunnarx.

rpannek commented:
Perhaps we can just revisit this in two weeks one your CI works again?
